### PR TITLE
Add eslint execution to linting

### PR DESCRIPTION
**Pull request by ssbarnea**
While we had eslint configured inside the repository, there was
not job ensuring that it did run. This change adds eslint as part
of our existing lint job (pre-commit).

Only two files were automatically reformatted by this change.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -60,7 +60,7 @@
     "@typescript-eslint/eslint-plugin": "^4.28.3",
     "@typescript-eslint/parser": "^4.28.3",
     "chai": "^4.3.4",
-    "eslint": "^7.30.0",
+    "eslint": "^7.32.0",
```
Comment (webknjaz) [inline package.json:63]: Why is this necessary?

File: package-lock.json [hunk 2]
```diff
@@ -31,15 +31,16 @@
         "@typescript-eslint/eslint-plugin": "^4.28.3",
         "@typescript-eslint/parser": "^4.28.3",
         "chai": "^4.3.4",
-        "eslint": "^7.30.0",
+        "eslint": "^7.32.0",
```
Comment (webknjaz) [inline package-lock.json:34]: Same question: https://github.com/ansible/ansible-language-server/pull/36/files#r711063087.

File: package-lock.json [hunk 3]
```diff
@@ -31,15 +31,16 @@
...
         "eslint-config-prettier": "^8.3.0",
         "mocha": "^8.4.0",
         "rimraf": "^3.0.2",
         "ts-node": "^9.1.1",
         "typescript": "^4.3.5"
       },
       "engines": {
-        "node": "*"
+        "node": ">=12.0",
```
Comment (webknjaz) [inline package-lock.json:42]: Why wasn't this added in previous lockfile updates?

File: package.json (hunk 1 above)
Comment (ssbarnea) [reply package.json:63]: Not yet sure as even if I had eslint 7 already installed I got an error about `es2021` not being valid in current config, upgrading did fix it but I am not sure which version I used before that.

File: package-lock.json (hunk 3 above)
Comment (ssbarnea) [reply package-lock.json:42]: I wondered the same when I seen it before uploading.

My guess is that yesterday we merged https://github.com/ansible/ansible-language-server/commit/1514f0fee1ba454c0cdc70473ae419dfe86a4fa5 we did not had a pipeline setup.

Comment (webknjaz) [reply package-lock.json]: I don't think so. Your last merge today doesn't fail: https://github.com/ansible/ansible-language-server/runs/3632442387. It would've failed if this was the case.

Comment (webknjaz) [reply package-lock.json]: I think the most plausible explanation is that `npm ci` probably only complains about the deps and not other metadata :man_shrugging:

**Review by webknjaz — APPROVED** [review]
